Messenger/AppDelegate.mm

```diff
@@ -302,15 +330,16 @@ - (void)webView:(WebView *)webView didFinishLoadForFrame:(WebFrame *)frame {
   auto rsp = frame.dataSource.response;
   if ([rsp isKindOfClass:[NSHTTPURLResponse class]] && ((NSHTTPURLResponse*)rsp).statusCode == 400) {
     NSLog(@"%@%@ frame.dataSource.response=%@", self, NSStringFromSelector(_cmd), frame.dataSource.response);
-    [webView.mainFrame.windowObject evaluateWebScript:
-     [NSString stringWithFormat:@"document.body.innerText = ''; var e = document.createElement('p'); document.body.appendChild(e); e.innerText = 'Oh noes. It appears Messenger.com is down for maintenance. Please try again later.'; var s = e.style; s.font='18px helvetica-light'; s.lineHeight='27px'; s.color='#999'; s.margin='0 auto'; s.width='50%%'; s.textAlign='center'; s.margin='0 auto'; s.marginTop='100px'; s.marginBottom='30px'; s.width='235px'; s.height='235px'; s.paddingTop='250px'; s.backgroundRepeat='no-repeat'; s.backgroundPosition='top center'; s.backgroundImage='url(%@)';", kErrorPNGDataURL]];
+    [self callJS:@"showMaintenanceMessage", nil];
+  } else {
+    [_webView.windowScriptObject evaluateWebScript:_injectionJS];
```

Comment:
This should be moved to `webView:didClearWindowObject:forFrame:` as to make it immediately available rather than after all HTML and resources have been loaded, which can take several minutes in worst-case scenarios.